lib/packages.js

```diff
@@ -379,6 +379,16 @@ function _sortPackages(cache) {
   const unsorted = Object.keys(cache).map(name => cache[name]);
   const sorted = [];
 
+  const cyclicDetectionThreshold = 10;
+  let maxCyclicCount = 0;
+  const cyclicDetector = {};
+
+  for(let i = 0; i < unsorted.length; i++) {
```

Comment:
there is a code convention for doing `++i` in for loops.

Reply:
Also consider `for...of` style:
```js
for(const pkg of unsorted) {
   const key = ...
   ...
}
```